src/dialect/oracle.rs

```diff
@@ -75,6 +82,35 @@ impl Dialect for OracleDialect {
         true
     }
 
+    fn get_next_precedence(&self, _parser: &Parser) -> Option<Result<u8, ParserError>> {
+        let t = _parser.peek_token();
+        debug!("get_next_precedence() {t:?}");
+
+        match t.token {
+            Token::StringConcat => Some(Ok(self.prec_value(Precedence::PlusMinus))),
+            _ => None,
+        }
+    }
+
+    fn prec_value(&self, prec: Precedence) -> u8 {
+        match prec {
+            Precedence::Period => 100,
+            Precedence::DoubleColon => 50,
+            Precedence::AtTz => 41,
+            Precedence::MulDivModOp => 40,
+            Precedence::PlusMinus => 30,
+            Precedence::Xor => 24,
+            Precedence::Ampersand => 23,
+            Precedence::Caret => 22,
+            Precedence::Pipe => 21,
+            Precedence::Between | Precedence::Eq | Precedence::Like | Precedence::Is => 20,
```

Comment:
This looks like the only changes compared to the default dialect precedence, can we add tests demonstrating what they imply?

Also since we're manually overriding all the precedence values, it would be good to have test coverage for each variant here for oracle as well